### PR TITLE
Fix Coverity issues

### PR DESCRIPTION
**Pull request by jrw972**
- 1565926
- 1565951
- 1565956
- 1565958
- 1565962
- 1565966
- 1565979
- 1565993
- 1565995
- 1566005
- 1567659
- 1567660
- 1573396
- 1573398
- 1573400
- 1573403
- 1573404
- 1573405
- 1573406

### CODE REVIEW

**Comment by mitza-oci** [comment]
Maybe not detected by Coverity (I haven't checked), but in #4339 it seems like the asserts in `bool test_equality(DDS::DynamicType_ptr lhs, DDS::DynamicType_ptr rhs, DynamicTypePtrPairSeen& dt_ptr_pair)` are now unreachable

File: dds/DCPS/security/AuthenticationBuiltInImpl.cpp [hunk 1]
```diff
@@ -1023,20 +1023,21 @@ ::CORBA::Boolean AuthenticationBuiltInImpl::return_identity_handle(
...
-    {
-      ACE_Guard<ACE_Thread_Mutex> handshake_data_guard(handshake_mutex_);
-
-      for (HandshakeDataMap::iterator it = handshake_data_.begin(); it != handshake_data_.end(); /* increment in loop*/) {
-        if (it->second.second == remote->second) {
-          handshake_data_.erase(it++);
-        } else {
-          ++it;
+    if (remote != local->second->validated_remotes.end()) {
+      {
```
Comment (mitza-oci) [inline dds/DCPS/security/AuthenticationBuiltInImpl.cpp:1027]:
```suggestion
    if (remote != local->second->validated_remotes.end()) {
```
with matching change below...

Comment (jrw972) [reply dds/DCPS/security/AuthenticationBuiltInImpl.cpp:1027]: There is one statement outside of the scoped lock.

Comment (mitza-oci) [reply dds/DCPS/security/AuthenticationBuiltInImpl.cpp:1027]: Right, I missed it.

**Review by mitza-oci — APPROVED** [review]

**Comment by jwillemsen** [comment]
After the merge of this PR I see a lot of test failures in one of our builds, see https://download.remedy.nl/buildlogs/rhel80_opendds_versioned/

**Comment by jwillemsen** [comment]
Did another build, now the test errors are gone, strange, maybe something hang on the jenkins slave, will keep an eye on our builds

**Pull request merged**